# Notebook to clean and organize data for input into Neural Net
The goal of this notebook is to get data organized to feed into neural network. The following need to be done:
- Remove videos from dataset
- Create subsets of data for each different insulation products, if the image is rotated, and if the image has a defect or not

## Imports and initializations

In [1]:
#Imports
import os

#Initializations
base = "Data"
all_data_base = "Data\\AllImages"

## Explore raw data 

In [2]:
#Read in directory of images
all_images = os.listdir(all_data_base)

#Check how many total image (movies and videos) we have
len(all_images)

436

__We have 668 images and videos (reads 436 now the code was rerun after removing movies)__

__How many images are present?__

In [3]:
#Look at number of images only
images = [file for file in all_images if file.split(".")[-1] == "JPG"]
len(images)

436

__We 436 images__

__Remove the movies from the directory__

In [4]:
#Get paths
paths = [os.path.join(all_data_base, file) for file in all_images]

#Grab only images 
for path in paths:
    if path.split(".")[-1] == "MOV":
        os.remove(path)

In [5]:
len(os.listdir(all_data_base))

436

__Now the AllImages directory contains only images of insulation__
## Some of the images got rotated during photographing. Rotate images such that all images are in the proper orientation.

In [9]:
#Imports
import cv2
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

In [13]:
#Take a look at the number of rows in each image
shapes = pd.DataFrame([cv2.imread(path).shape[1] for path in paths])
shapes.value_counts()

3024    308
4032    128
dtype: int64

__Two heights of images (3024 px and 4032 px). 308/436 images are 3024 px wide - reshape all images to 4032 px x 3024 px__

Rotate to these dimensions since it is easier to take 4032 x 3024 px images with one hand on cell phone

In [26]:
def reshape_save(paths):
    for path in paths:
        img = cv2.imread(path)
        if img.shape[1] == 4032:
            cv2.imwrite(path, cv2.rotate(img, cv2.ROTATE_90_CLOCKWISE))

In [27]:
reshape_save(paths)

#### All images shapes should be 4032 x 3024

In [30]:
#Take a look at height of each image
paths = [os.path.join(all_data_base, file) for file in all_images]
shapes = pd.DataFrame([cv2.imread(path).shape[0] for path in paths])
shapes.value_counts()

4032    436
dtype: int64

__We have 436 images which have a height of 4032 pixels. Thus, all images are now 4032 x 3024 px__